lib/src/main/java/com/smileidentity/compose/consent/bvn/BvnInputScreen.kt

```diff
@@ -0,0 +1,148 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material3.Button
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.smileidentity.R
+import com.smileidentity.compose.components.SmileIDAttribution
+import com.smileidentity.compose.preview.Preview
+import com.smileidentity.compose.preview.SmilePreviews
+import com.smileidentity.viewmodel.BvnConsentViewModel
+import com.smileidentity.viewmodel.viewModelFactory
+
+@Composable
+fun BvnInputScreen(
+    cancelBvnVerification: () -> Unit,
+    onBvnVerified: () -> Unit,
+    modifier: Modifier = Modifier,
+    showAttribution: Boolean = true,
+    viewModel: BvnConsentViewModel = viewModel(
+        factory = viewModelFactory {
+            BvnConsentViewModel()
+        },
+    ),
+) {
+    val uiState = viewModel.uiState.collectAsStateWithLifecycle().value
+    var bvn by rememberSaveable { mutableStateOf("") }
+
+    val focusRequester = remember { FocusRequester() }
+
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .padding(24.dp),
+    ) {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            Text(
+                text = stringResource(id = R.string.si_bvn_enter_id_title),
+                textAlign = TextAlign.Center,
+                style = MaterialTheme.typography.headlineMedium,
+                fontWeight = FontWeight.ExtraBold,
+                modifier = Modifier.weight(1f),
+            )
+            Icon(
+                imageVector = Icons.Default.Close,
+                contentDescription = null,
+                modifier = Modifier
+                    .testTag("bvn_input_screen_cancel")
+                    .clickable { cancelBvnVerification() },
```

Comment:
nit, we can pass method reference (should be better performance, albeit very minimal)?
```suggestion
                    .clickable(::cancelBvnVerification),
```